File: src/cf/contest/1541/b/b.py
#    Created by rdstihz at 2021-07-02 20:20:52

T = int(input())

for k in range(T):
    n = int(input())
    a = [0]
    a.extend(map(int, input().split()))
    ans = 0
    for i in range(1, n + 1):
        st = ((a[i] - i) % a[i] + a[i]) % a[i]
        for j in range(st, i, a[i]):
            if i + j == a[i] * a[j]:
                ans += 1
    print(ans)

File: src/cf/contest/1541/b/b.cpp
/*
    Created by rdstihz at 2021-06-26 00:04:39
*/

#include <bits/stdc++.h>

using namespace std;

const int maxn = 100000 + 100;

int a[maxn];

int main() {

    int T;
    scanf("%d", &T);
    while (T--) {
        int n;
        scanf("%d", &n);
        for (int i = 1; i <= n; i++)
            scanf("%d", a + i);

        long long ans = 0;

        for (int i = 1; i <= n; i++) {
            int t = ((a[i] - i) % a[i] + a[i]) % a[i];
            for (int j = t; j < i; j += a[i])
                if (i + j == a[i] * a[j])
                    ans++;
        }

        printf("%lld\n", ans);
    }

    return 0;
}